projects/gloo/cli/pkg/cmd/install/knative.go

```diff
@@ -0,0 +1,320 @@
+
+	knativeCrdSelector = "knative.dev/crd-install=true"
+
+	servingReleaseUrl    = "https://github.com/knative/serving/releases/download/v%v/serving.yaml"
+	buildReleaseUrl      = "https://github.com/knative/build/releases/download/v%v/build.yaml"
+	eventingReleaseUrl   = "https://github.com/knative/eventing/releases/download/v%v/release.yaml"
+	monitoringReleaseUrl = "https://github.com/knative/serving/releases/download/v%v/monitoring.yaml"
+
+	knativeIngressProviderLabel = "networking.knative.dev/ingress-provider"
+	knativeIngressProviderIstio = "istio"
+
+	yamlJoiner = "\n---\n"
+)
+
+func knativeCmd(opts *options.Options) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "knative",
+		Short: "install Knative with Gloo on kubernetes",
+		Long:  "requires kubectl to be installed",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Install.Knative.InstallKnative {
+				installed, _, err := checkKnativeInstallation()
+				if err != nil {
+					return errors.Wrapf(err, "checking for existing knative installation")
+				}
+				if installed {
+					return errors.Errorf("knative-serving namespace found. please " +
+						"uninstall the previous version of knative, or re-run this command with --install-knative=false")
+				}
+
+				if err := installKnativeServing(opts); err != nil {
+					return errors.Wrapf(err, "installing knative components failed. "+
+						"options used: %#v", opts.Install.Knative)
+				}
+			}
+
+			if !opts.Install.Knative.SkipGlooInstall {
+				if err := installGloo(opts, constants.KnativeValuesFileName); err != nil {
+					return errors.Wrapf(err, "installing gloo in knative mode")
+				}
+			}
+			return nil
+		},
+	}
+	pflags := cmd.PersistentFlags()
+	flagutils.AddInstallFlags(pflags, &opts.Install)
+	flagutils.AddKnativeInstallFlags(pflags, &opts.Install.Knative)
+	return cmd
+}
+
+func installKnativeServing(opts *options.Options) error {
+	knativeOpts := opts.Install.Knative
+
+	// store the opts as a label on the knative-serving namespace
+	// we can use this to uninstall later on
+	knativeOptsJson, err := json.Marshal(knativeOpts)
+	if err != nil {
+		return err
+	}
+
+	manifests, err := RenderKnativeManifests(knativeOpts)
+	if err != nil {
+		return err
+	}
+	if opts.Install.DryRun {
+		fmt.Printf("%s", manifests)
+		// For safety, print a YAML separator so multiple invocations of this function will produce valid output
+		fmt.Printf(yamlJoiner)
+		return nil
+	}
+
+	// install crds first by using selector
+	// https://knative.dev/docs/install/knative-with-any-k8s/
+	fmt.Fprintln(os.Stderr, "installing Knative CRDs...")
+	if err := install.KubectlApply([]byte(manifests), "--selector", knativeCrdSelector); err != nil {
+		return errors.Wrapf(err, "installing knative crds with kubectl apply")
+	}
+
+	fmt.Fprintln(os.Stderr, "installing Knative...")
+	if err := install.KubectlApply([]byte(manifests)); err != nil {
+		return errors.Wrapf(err, "installing knative crds with kubectl apply")
```

Comment:
```suggestion
		return errors.Wrapf(err, "installing knative resources with kubectl apply")
```
Or something similar.